docs/book/start/getting-data-to-user-space.md

```diff
@@ -0,0 +1,174 @@
+# Getting data to user-space
+
+In previous chapters we were logging packets with aya-log. However, what if we
+need to send additional data about the packet or any other type of information
+for the user-space program to utilize? In this chapter, we will explore how to
+leverage perf buffers and define data structures in the *-common* crate to
+transfer data from eBPF program to user-space applications. By doing so,
+user-space programs can access and utilize the transferred data effectively.
+
+!!! example "Source code"
+
+    Full code for the example in this chapter is available
+    [here](https://github.com/aya-rs/book/tree/main/examples/xdp-perfbuf-custom-data)
+
+## Sharing data
+
+In this chapter, we will be sending data from the kernel space to the user space
+by writing it into a struct and outputting it to user space. We can achieve this
+by using an eBPF map. There are different types of maps available, but in this
+case, we will use `PerfEventArray`.
+
+`PerfEventArray` is a collection of per-CPU circular buffers that enable the
+kernel to emit events (defined as custom structs) to user space. Each CPU has
+its own buffer, and the eBPF program emits an event to the buffer of the CPU
+it's currently running on. The events are out of order, meaning that they arrive
```

Comment:
I think the events are **unordered** - I don't think a feature of PerfEventArray is delivering events always out of order 😂